frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-internal/status-card/status-card.tsx

```diff
@@ -74,6 +80,75 @@ export const StatusCard: React.FC<DashboardItemProps> = ({
     t,
   );
 
+  const [isHCPresent, setHCPresence] = React.useState(false);
+  const healthChecksGetterOrPresent = (isPresent: boolean = false) => {
+    const cephDetails = data?.[0]?.status?.ceph?.details;
+    const pattern = /[A-Z]+_*|error/g;
+    const healthChecks: CephHealthCheck[] = [];
+    for (const key in cephDetails) {
+      if (pattern.test(key)) {
+        if (isPresent) {
+          return true;
+        }
+        const healthCheckObject: CephHealthCheck = {
+          details: cephDetails[key].message,
+          troubleshootLink: whitelistedHealthChecksRef[key] ?? null,
+        };
+        healthChecks.push(healthCheckObject);
+      }
+    }
+    return isPresent ? false : healthChecks;
+  };
+
+  React.useEffect(
+    () => {
+      setHCPresence(healthChecksGetterOrPresent(true) as boolean);
```

Comment:
you can receive a health check array and decide true or false based on an array length or !! array 

returning different types from the same function is  risky

Reply:
```suggestion
      setHCPresence(!! healthChecksGetterOrPresent());
```

Reply:
I actually modified that function to only run once if `isPresent` argument that's passed is truthy, otherwise we'll have to create the whole array which isn't really needed in this usecase.